src/snoowrap.js

```diff
@@ -695,270 +728,374 @@ const snoowrap = class snoowrap {
   }
+
+  /**
+   * @TODO Modmail; Not implemented
+   */
+  getNewModmailConversations (options = {}) {
+    return this._getListing({
+      uri: 'api/mod/conversations', qs: options, _name: 'ModmailConversation', _transform: response => {
+        response.after = null;
+        response.before = null;
+        response.children = [];
+
+        for (const conversation of response.conversationIds) {
+          response.conversations[conversation].participant = this._newObject('ModmailConversationAuthor', {
+            ...response.conversations[conversation].participant
+          });
+          const conversationObjects = objects.ModmailConversation._getConversationObjects(
+            response.conversations[conversation],
+            response
+          );
+          const data = {
+            ...conversationObjects,
+            ...response.conversations[conversation]
+          };
+          response.children.push(this._newObject('ModmailConversation', data));
+        }
+        return this._newObject('Listing', response);
+      }
+    });
+  }
+
+  /**
+   * @summary Creates a new moderator discussion
+   * @param {object} options
+   * @param {string} options.body Body of the discussion
+   * @param {string} options.subject Title or subject
+   * @param {string} options.srName Subreddit name without fullname
+   */
+  createModeratorDiscussion ({
+                               body,
+                               subject,
+                               srName
+                             }) {
+    const parsedFromSr = srName.replace(/^\/?r\//, ''); // Convert '/r/subreddit_name' to 'subreddit_name'
+    // _newObject ignores most of the response, no practical way to parse the returned content yet
+    return this._post({
+      uri: 'api/mod/conversations', form: {
+        body, subject, srName: parsedFromSr
+      }
+    }).then(res => this._newObject('ModmailConversation', {id: res.conversation.id}));
+  }
+
+  getNewModmailConversation (id) {
+    return this._newObject('ModmailConversation', {id});
+  }
+
+  markNewModmailConversationsAsRead (conversations) {
+    const conversationIds = conversations.map(message => addFullnamePrefix(message, ''));
+    return this._post({uri: 'api/mod/conversations/read', form: {conversationIds: conversationIds.join(',')}});
+  }
+
+  markNewModmailConversationsAsUnread (conversations) {
+    const conversationIds = conversations.map(message => addFullnamePrefix(message, ''));
+    return this._post({uri: 'api/mod/conversations/unread', form: {conversationIds: conversationIds.join(',')}});
+  }
+
+  getNewModmailSubreddits () {
+    return this._getListing({
+      uri: 'api/mod/conversations/subreddits', _transform: response => {
+        return this._newObject('Listing', {
+          after: null,
+          before: null,
+          children: Object.values(response.subreddits).map(s => {
+            return this._newObject('Subreddit', s);
+          })
+        });
+      }
+    });
+  }
```

Comment:
@not-an-aardvark Is there a better way to do this? The format received from the endpoint is as follows:

```json
{
  "subreddits": {
    "t5_xogi3": {
      "display_name": "SpyTecSnoowrapTesting",
      "name": "SpyTecSnoowrapTesting",
      "lastUpdated": "2019-03-19T09:45:13.994306+00:00",
      "keyColor": "",
      "subscribers": 1,
      "id": "t5_xogi3",
      "icon": ""
    },
    "t5_xog8s": {
      "display_name": "u/SpyTecSnoowrap",
      "name": "u_SpyTecSnoowrap",
      "lastUpdated": null,
      "keyColor": "",
      "subscribers": 0,
      "id": "t5_xog8s",
      "icon": "https://www.redditstatic.com/avatars/avatar_default_20_46D160.png"
    }
  }
}
```

Reply:
If it's not paginated, would it make sense to just use an array instead of a Listing?

Reply:
That makes sense, I’ll change it when I get the chance

Reply:
Changed to be an array, tests have been updated as well